tap-gui/configurator/external-plugins.hbs.md

```diff
@@ -11,4 +11,597 @@ own private registry or a plug-in registry if you intend to use a third-party or
 
 > **Important** Tanzu Application Platform plug-ins cannot be removed from customized portals.
 > However, if you decide you want to hide them, you can use the
-> [runtime configuration](concepts.hbs.md#runtime) options in your `tap-values.yaml` file.
\ No newline at end of file
+> [runtime configuration](concepts.hbs.md#runtime) options in your `tap-values.yaml` file.
+
+## Extracting the contents of the builder image
+
+Find the location of the builder image by querying your TAP cluster
+
+```shell
+$ kubectl -n tap-install get packages tpb.tanzu.vmware.com.0.1.2 -o jsonpath={.spec.template.spec.fetch[0].imgpkgBundle.image}
+```
+
+```
+REGISTRY/REPOSITORY@sha256:458d9727b6f77ea2fadd2ee597885419e555e7b74057bd654f7d4eee8f006641
+```
+
+This is a carvel bundle containing the builder image, in order to get the
+address of the builder image you must download the carvel bundle and look at the
+image lock file within.
+
+```shell
+$ imgpkg pull -b REGISTRY/REPOSITORY@sha256:458d9727b6f77ea2fadd2ee597885419e555e7b74057bd654f7d4eee8f006641 -o bundle
+$ cat bundle/.imgpkg/images.yml
+```
+
+```
+---
+apiVersion: imgpkg.carvel.dev/v1alpha1
+images:
+- annotations:
+    kbld.carvel.dev/id: harbor-repo.vmware.com/esback/tdp-configurator@sha256:0b03767e0526803c055d080a7d378df9adaeaa7cff0ff5af23a2debe668f20e3
+  image: REGISTRY/REPOSITORY@sha256:0b03767e0526803c055d080a7d378df9adaeaa7cff0ff5af23a2debe668f20e3
+kind: ImagesLock
+```
+
+We want to download the image specified by the `image` key.
+
+```shell
+$ imgpkg -i REGISTRY/REPOSITORY@sha256:0b03767e0526803c055d080a7d378df9adaeaa7cff0ff5af23a2debe668f20e3 -o builder
+```
+
+Will output the filesystem of the builder image to a directory called `builder` on your local machine.
+
+## Running a local verdaccio server for inner loop development
+
+Using your extracted builder from the previous step you can run a
+[verdaccio](https://verdaccio.org/) server on your machine. This will enable you
+to depend on typescript libraries that the Tanzu Developer Portal has developed
+and are necessary for adding plugins to your Tanzu Developer Portal instance.
+
+note: as of writing this doc a workaround is needed to adjust the parameters
+of the verdaccio `offline_config.yaml`. In order to fetch recently published
+packages you can adjust the `maxage` parameter under the `npmjs` `uplinks`
```

Comment:
We don't expect the workaround to become obsolete in TAP 1.6.1. Is this note necessary?

Reply:
This workaround would still be necessary in the 1.6 timeframe. Actually it's a bit of a blocker in more ways than one :cry:. This could also potentially block customers from pulling in custom plugin wrappers from NPM, since the builder image won't reach out to NPM to get metadata on new packages